diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -193,6 +193,18 @@ fun ASTNode.findAllNodesWithSpecificType(elementType: IElementType): List<ASTNod
     }
 }
 
+/**
+ * This method recursively finds parent node that has specified elementType
+ */
+fun ASTNode.findParentNodeWithSpecificType(elementType: IElementType): ASTNode? {
+    if (this.elementType == elementType) return this
+    if (this.treeParent == null) {
+        return null
+    } else {
+        return this.treeParent.findParentNodeWithSpecificType(elementType)
```

Comment:
Isn't `ASTNode.parents()` from kotlin compiler which returns `sequence` better than recursion?

Reply:
thank you so much, didn't know about that method!